base/applications/cleanmgr/util.c

```diff
@@ -0,0 +1,1179 @@
+    
+    if (!RegValSet(ValStr, L"SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Explorer\\VolumeCaches\\Recycle Bin", bv.RecycleClean))
+        return;
+}
+
+void SagerunProc(int nArgs, PWCHAR ArgReal, LPWSTR* ArgList, PWCHAR LogicalDrives)
+{
+    WCHAR *ValStr = NULL;
+    WCHAR* SingleDrive = LogicalDrives;
+    INT_PTR DialogButtonSelect;
+        
+    ValStr = RealStageFlag(nArgs, ArgReal, ArgList);
+        
+    if (RegQuery(ValStr, L"SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Explorer\\VolumeCaches\\Old ChkDsk Files") == ALLOW_FILE_REMOVAL)
+        bv.ChkDskClean = TRUE;
+        
+    if (RegQuery(ValStr, L"SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Explorer\\VolumeCaches\\Temporary Files") == ALLOW_FILE_REMOVAL)
+        bv.TempClean = TRUE;
+
+    if (RegQuery(ValStr, L"SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Explorer\\VolumeCaches\\RAPPS Files") == ALLOW_FILE_REMOVAL)
+        bv.RappsClean = TRUE;
+        
+    if (RegQuery(ValStr, L"SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Explorer\\VolumeCaches\\Recycle Bin") == ALLOW_FILE_REMOVAL)
+        bv.RecycleClean = TRUE;
+        
+    while (*SingleDrive)
+    {
+        if (GetDriveTypeW(SingleDrive) == ONLY_PHYSICAL_DRIVE)
+        {
+            StringCchCopyW(wcv.DriveLetter, MAX_PATH, SingleDrive);
+            wcv.DriveLetter[wcslen(wcv.DriveLetter) - 1] = '\0';
+
+            DialogButtonSelect = DialogBoxParamW(NULL, MAKEINTRESOURCEW(IDD_PROGRESS), NULL, ProgressDlgProc, 0);
+
+            if(DialogButtonSelect == IDCANCEL)
+                return;
+        
+            DialogBoxParamW(NULL, MAKEINTRESOURCEW(IDD_PROGRESS_END), NULL, ProgressEndDlgProc, 0);
+                
+            ZeroMemory(&wcv.DriveLetter, sizeof(wcv.DriveLetter));
+        }
+        SingleDrive += wcslen(SingleDrive) + 1;
+    }
+}
+
+void InitListViewControl(HWND hList)
+{
+    WCHAR TempList[MAX_PATH] = { 0 };
+
+    LVCOLUMNW lvC;
+    ZeroMemory(&lvC, sizeof(lvC));
+    lvC.mask = LVCF_WIDTH | LVCF_SUBITEM | LVCF_FMT;
+    lvC.cx = 158;
+    lvC.cchTextMax = 256;
+    lvC.fmt = LVCFMT_RIGHT;
+
+    SendMessageW(hList, LVM_SETEXTENDEDLISTVIEWSTYLE, 0, LVS_EX_CHECKBOXES | LVS_EX_FULLROWSELECT);
+
+    ListView_InsertColumn(hList, 0, &lvC);
+    ListView_InsertColumn(hList, 1, &lvC);
+
+    if (!CreateImageLists(hList))
+    {
+        MessageBoxW(NULL, L"CreateImageLists() failed!", L"Error", MB_OK | MB_ICONERROR);
+    }
+
+    StringCchPrintfW(TempList, _countof(TempList), L"%.02lf %s", SetOptimalSize(sz.ChkDskSize), FindOptimalUnit(sz.ChkDskSize));
+    AddItem(hList, L"Old ChkDsk Files", TempList, 0);
+
+    StringCchPrintfW(TempList, _countof(TempList), L"%.02lf %s", SetOptimalSize(sz.RappsSize), FindOptimalUnit(sz.RappsSize));
+    AddItem(hList, L"RAPPS Files", TempList, 0);
+ 
+    StringCchPrintfW(TempList, _countof(TempList), L"%.02lf %s", SetOptimalSize(sz.RecycleBinSize), FindOptimalUnit(sz.RecycleBinSize));
+    AddItem(hList, L"Recycle Bin", TempList, 2);
```

Comment:
I suppose that these labels should be retrieved from somewhere else, and be localized?